.github/workflows/cd.yml

```diff
@@ -24,22 +23,12 @@ jobs:
 
       - name: Generate Tag
         id: tag
-        env:
-          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-          GH_REPO: ${{ github.repository }}
-        # We use the `gh` utility as it offers much easier traversal of GitHub-
-        # related things (prs, issues, etc...).
-        # In this merged_branch variable, we get the list of PRs that have been merged,
-        # and sort them by when they were merged, taking the most recent (last) and
-        # returning the branch name of that PR.
-        # The `cut` command splits the `pre-<version>` branch into just `<version>`,
+        # Get the tag name from the access-om2 spec in the `spack.yaml`.
+        # The `cut` command splits the `access-om2@git.<version>` field into just `<version>`,
         # which will be the new tag.
         run: |
-          merged_branch=$(gh pr list \
-            --state merged \
-            --json 'headRefName,mergedAt' \
-            --jq 'sort_by(.mergedAt) | last | .headRefName')
-          echo "name=$(cut --delimiter '-' --fields 2 <<< '$merged_branch')" >> $GITHUB_OUTPUT
+          access-om2-package=$(yq e '.spack.specs[0]' spack.yaml)
+          echo "name=$(cut --delimiter '.' --field 2- <<< '${access-om2-package}')" >> $GITHUB_OUTPUT
```

Comment:
Woah, I never knew about that format for substitution!
And yes we can be sure - if you don't specify a `shell:` in the step, it will be `bash` (since it `runs-on: ubuntu-latest`). Alternatively we can just do `shell: bash` everywhere or set it as a default.